src/EditorFeatures/Test/Diagnostics/DiagnosticAnalyzerServiceTests.cs

```diff
@@ -276,13 +276,17 @@ public void TestHostAnalyzerOrdering()
             var incrementalAnalyzer = (DiagnosticIncrementalAnalyzer)service.CreateIncrementalAnalyzer(workspace);
             var analyzers = incrementalAnalyzer.GetAnalyzersTestOnly(project).ToArray();
 
-            Assert.Equal(typeof(CSharpCompilerDiagnosticAnalyzer), analyzers[0].GetType());
-            Assert.Equal(typeof(Analyzer), analyzers[1].GetType());
-            Assert.Equal(typeof(Priority0Analyzer), analyzers[2].GetType());
-            Assert.Equal(typeof(Priority1Analyzer), analyzers[3].GetType());
-            Assert.Equal(typeof(Priority10Analyzer), analyzers[4].GetType());
-            Assert.Equal(typeof(Priority15Analyzer), analyzers[5].GetType());
-            Assert.Equal(typeof(Priority20Analyzer), analyzers[6].GetType());
+            AssertEx.Equal(new[]
+            {
+                typeof(FileContentLoadAnalyzer),
+                typeof(CSharpCompilerDiagnosticAnalyzer),
+                typeof(Analyzer),
+                typeof(Priority0Analyzer),
+                typeof(Priority1Analyzer),
+                typeof(Priority10Analyzer),
+                typeof(Priority15Analyzer),
+                typeof(Priority20Analyzer)
+            }, analyzers.Select(a => a.GetType()));
```

Comment:
💡 Consider using the `EqualOrDiff` form I used in #40097. It provides a much better message if the order is wrong or items are added/removed.

Reply:
`AssertEx.Equal` also displays a diff and works with sequences, unlike `EqualOrDiff` that works with strings.

Reply:
I'm specifically referring to this:

https://github.com/dotnet/roslyn/blob/59583144b3b03563151e834dbae8cf70852e4c20/src/EditorFeatures/CSharpTest/Completion/CompletionProviders/CompletionProviderOrderTests.cs#L70-L72

Reply:
You can use `AssertEx.Equal` there to avoid the string joins.

Reply:
We should update `AssertEx.Equal` to use DiffPlex like `EqualOrDiff`. Currently it produces a diff message but it's been problematic in practice for me.

Reply:
📝 I missed that you were using `AssertEx.Equal` here as opposed to `Assert.Equal`, which also exists for collection inputs but doesn't have any diff support at all.